app/src/main/kotlin/com/wire/android/util/UriUtil.kt

```diff
@@ -28,9 +30,33 @@ fun containsSchema(url: String): Boolean {
 }
 
 fun normalizeLink(url: String): String {
-    return if (containsSchema(url)) {
-        url
+    val sanitizedUrl = sanitizeUrl(url)
+    return if (containsSchema(sanitizedUrl)) {
+        sanitizedUrl
     } else {
-        "https://$url"
+        "https://$sanitizedUrl"
+    }
+}
+
+@Suppress("TooGenericExceptionCaught")
+fun sanitizeUrl(url: String): String {
+    try {
+        val urlComponents = url.split("://")
+        val scheme = urlComponents[0] // Extract the URL scheme (e.g., "http")
+        val restOfUrl = urlComponents[1] // Extract the rest of the URL
+
+        // Split the rest of the URL by '/' to isolate the domain
+        val domainAndPath = restOfUrl.split("/")
+        val domain = domainAndPath[0] // Extract the domain
+
+        // Use IDN.toASCII to convert the domain to ASCII representation
+        val asciiDomain = IDN.toASCII(domain)
+
+        // Reconstruct the sanitized URL
+        return "$scheme://$asciiDomain/${domainAndPath.subList(1, domainAndPath.size).joinToString("/")}"
+    } catch (e: Exception) {
+        // Handle any exceptions that might occur during the processing
+        appLogger.e("Error sanitizing URL: $url", e)
```

Comment:
```suggestion
        appLogger.w("Error sanitizing URL: $url", e)
```
Suggestion: Minor nitpick, since we are going to fallback the url, maybe a warning suits better